src/components/Tool/Import/Csv/CsvHeaders.tsx

```diff
@@ -0,0 +1,321 @@
+import { useApiConstants } from 'src/components/Constants/UseApiConstants';
+import { Confirmation } from 'src/components/common/Modal/Confirmation/Confirmation';
+import useGetAppSettings from 'src/hooks/useGetAppSettings';
+import {
+  CsvImportContext,
+  CsvImportType,
+  CsvImportValue,
+  CsvImportViewStepEnum,
+} from './CsvImportContext';
+import { HeaderBox } from './HeaderBox';
+import { get, save } from './csvImportService';
+import { useRequiredHeaders, useSupportedHeaders } from './uploadCsvFile';
+
+export interface CsvHeadersProps {
+  accountListId: string;
+  setCurrentTab: (currentTab: CsvImportViewStepEnum) => void;
+}
+
+const buildDefaultFileHeadersMappings = (
+  fileHeadersMappings: object,
+  uploadData: CsvImportType | null,
+  supportedHeaders: object,
+): object => {
+  const defaultMappings = {};
+
+  if (!Object.keys(fileHeadersMappings).length && uploadData?.fileHeaders) {
+    Object.keys(uploadData.fileHeaders).forEach((header) => {
+      if (supportedHeaders[header]) {
+        defaultMappings[header] = header;
+      } else {
+        defaultMappings[header] = -1;
+      }
+    });
+  }
+  return defaultMappings;
+};
+
+const updateUnmappedHeaders = (
+  requiredHeaders: string[],
+  fileHeadersMappings: object,
+  setUnmappedHeaders: React.Dispatch<React.SetStateAction<string[]>>,
+) => {
+  const unmapped = requiredHeaders.filter((header) => {
+    return !Object.values(fileHeadersMappings).includes(header);
+  });
+  setUnmappedHeaders(unmapped);
+};
+
+const CsvHeaders: React.FC<CsvHeadersProps> = ({
+  accountListId,
+  setCurrentTab,
+}) => {
+  const { uploadData, setUploadData, initialData, setInitialData, csvFileId } =
+    useContext(CsvImportContext) as CsvImportValue;
+
+  const supportedHeaders = useSupportedHeaders();
+  const requiredHeaders = useRequiredHeaders();
+  const constants = useApiConstants();
+  const [unmappedHeaders, setUnmappedHeaders] =
+    useState<string[]>(requiredHeaders);
+  const [mappedHeaders, setMappedHeaders] = useState<string[]>([]);
+  const [showBackWarningModal, setShowBackWarningModal] = useState(false);
+  const { t } = useTranslation();
+  const { appName } = useGetAppSettings();
+
+  const [unmappedHeadersChecked, setUnmappedHeadersChecked] = useState(false);
+
+  const importHeaders = uploadData?.fileHeaders ?? {};
+  const fileHeadersMappings = uploadData?.fileHeadersMappings ?? {};
+
+  const updateHeaders = (uploadData) => {
+    updateUnmappedHeaders(
+      requiredHeaders,
+      uploadData.fileHeadersMappings,
+      setUnmappedHeaders,
+    );
+
+    const mapped = requiredHeaders.filter((header) => {
+      return Object.values(uploadData.fileHeadersMappings).includes(header);
+    });
+    setMappedHeaders(mapped);
+  };
+
+  useEffect(() => {
+    if (uploadData?.id) {
+      if (!initialData?.id) {
+        get(accountListId, uploadData.id, initialData).then((data) => {
+          setInitialData(data);
+        });
```

Comment:
It's best to use await then `.then`. But I don't want to slow you down, so I'm good with this going live no and we come back later to change it.

Reply:
I don't think a `useEffect` can be an async function.